From fb769c380eeacf23b64cb3bf7860ba49a0e0eed5 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 8 Apr 2025 00:15:57 -0700
Subject: [PATCH] pppd: Fix building session.c with GCC 15 (#553)

Fixed building with GCC 15 which defaults to C23
and find conflicting declration of getspnam() here
with the one provided by shadow.h (extern struct spwd *getspnam (const char *__name);)

Fixes
../../ppp-2.5.2/pppd/session.c: In function 'session_start':
../../ppp-2.5.2/pppd/session.c:185:18: error: conflicting types for 'getspnam'; have 'struct spwd *(void)'
  185 |     struct spwd *getspnam();
      |                  ^~~~~~~~

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream: https://github.com/ppp-project/ppp/commit/fb769c380eeacf23b64cb3bf7860ba49a0e0eed5

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 pppd/session.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pppd/session.c b/pppd/session.c
index f08d8e19..9cc75389 100644
--- a/pppd/session.c
+++ b/pppd/session.c
@@ -182,7 +182,6 @@ session_start(const int flags, const char *user, const char *passwd, const char
     char *cbuf;
 #ifdef HAVE_SHADOW_H
     struct spwd *spwd;
-    struct spwd *getspnam();
     long now = 0;
 #endif /* #ifdef HAVE_SHADOW_H */
 #endif /* #ifdef PPP_WITH_PAM */
